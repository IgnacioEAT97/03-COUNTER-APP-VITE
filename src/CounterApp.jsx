import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({value}) =>{

    const [counter, setCounter] = useState(value) // Cuando se cambia un hook, se vuelve a renderizar todo el functional component en el cual estaba incluido
    const handleAdd = () => {
    //console.log(event)
    //setCounter(counter + 1);
    setCounter((c) => c + 1 ); //Toma el valor actual del counter, sin saber que es el counter y le suma 1
    }

    const handleRes = () => setCounter(counter - 1)
    

    const handleReset = () => setCounter(value)

    return (
        <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>

        <button onClick={ handleAdd }> +1 </button>
        <button onClick={ handleRes }> -1 </button>
        <button aria-label='btn-reset' onClick={ handleReset}> Reset </button>
        </>
    )
}

CounterApp.propTypesropTypes= {
    value: PropTypes.number.isRequired
}