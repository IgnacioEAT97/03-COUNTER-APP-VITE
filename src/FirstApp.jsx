//import {Fragment} from 'react'
import PropTypes from 'prop-types'
const newMessage = {
  id: '123',
  pass: 'wenaa',
} // Si una coonstante no va a tener dependencia, se recomienda ponerla fuera del functional component, ya que no es necesario que se rerenderize
const Tarea = () => 'Mi nombre es Tarea';


export const FirstApp = ({title,subTitle,name}) => { //Props con desestructuracion

    return (
      <>{/* Esto es igual a un <Fragment>, fue a√±adido por react para no tener que importar cada vez el fragment */}
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      {/* <h1>{Tarea()}</h1> */}
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{name}</p>
      </>
    )
  }
  
  FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
  }

  FirstApp.defaultProps = { //Entran antes que los propTypes
    //title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Ignacio Alvarado',
  }