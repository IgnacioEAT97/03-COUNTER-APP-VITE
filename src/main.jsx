import React from 'react';
import ReactDOM from 'react-dom/client';
//import {HelloWorldApp} from './HelloWorldApp'
import {FirstApp} from './FirstApp'
import { CounterApp } from './CounterApp';

import './style.css'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <HelloWorldApp/> */}
        {/* <FirstApp title='Hola soy vegeta' /> */}   {/* Pasamos las props del componente padre al hijo  */}
         <CounterApp value={30}></CounterApp> 
    </React.StrictMode>
)