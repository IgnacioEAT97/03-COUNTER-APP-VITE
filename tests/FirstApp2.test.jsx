import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en componente FirstApp', () => { 
   const title = 'Hola soy goku'
   const subTitle = 'Soy un subtitulo'
   test('debe de hacer match con el snaoshot', () => { 

      const {container} = render(<FirstApp title={title}></FirstApp>)
      expect(container).toMatchSnapshot();
   });

   test('debe de mostrar el mensaje "Hola soy goku', () => { 
      
      render(<FirstApp title={title}></FirstApp>)
      //screen.debug(); //La ultima version actualizada antes de los cambios del DOM
      expect(screen.getByText(title)).toBeTruthy();
    });

    test('debe mostrar el titulo en un h1', () => { 
      render(<FirstApp title={title}></FirstApp>)
      expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
     });

     test('debe de mostrar el subtitulo enviado por props', () => {
      render(<FirstApp title={title} subTitle={subTitle}></FirstApp>)
      expect(screen.getAllByText(subTitle).length).toBe(2);
     })
});