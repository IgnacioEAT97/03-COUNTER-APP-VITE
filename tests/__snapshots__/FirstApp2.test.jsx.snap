// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente FirstApp debe de hacer match con el snaoshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Ignacio Alvarado
  </p>
</div>
`;
